### PR TITLE
Support sending image data as part of a user message, using a new ImageUrl.load() method. Add sample and test.

### PR DESCRIPTION
**Pull request by dargilco**
In a previous PR I had samples for sending up image URL as part of chat completions user message. In this PR I add a sample that shows how you can base-64 encode image data (ready from a file) and send it up. Hand crafted a class method ImageUrl.load() to better support this scenario. Tested using Phi3-v model.

### CODE REVIEW

File: sdk/ai/azure-ai-inference/samples/sample_chat_completions_with_image_data.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                    ),
+                ],
+            ),
+        ],
+    )
+
+    print(response.choices[0].message.content)
+
+
+def get_image_data_url(image_file: str, image_format: str) -> str:
```
Comment (johanste) [inline sdk/ai/azure-ai-inference/samples/sample_chat_completions_with_image_data.py:78]: I would love to not have to write this code as a user of our library. Any reason to not include an implementation in the library?

Comment (dargilco) [reply sdk/ai/azure-ai-inference/samples/sample_chat_completions_with_image_data.py:78]: I updated my "to do" feature list with "discuss accepting image file path directly and image type as input to ImageUrl() class when constructing a user message". I'll look into this in a separate PR.

Comment (dargilco) [reply sdk/ai/azure-ai-inference/samples/sample_chat_completions_with_image_data.py:77]: I added a constructor on ImageUrl to accept image file and image format. See latest commit.

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py [hunk 2]
```diff
@@ -7,18 +7,73 @@
...
+        """
+        :param mapping: raw JSON to initialize the model.
+        :type mapping: Mapping[str, Any]
+        """
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        image_file = kwargs.pop("image_file", None)
+        image_format = kwargs.pop("image_format", None)
+        if image_file and image_format:
+            with open(image_file, "rb") as f:
```
Comment (johanste) [inline sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py:71]: You should make sure that the caller didn't also specify `url` (and raise a `TypeError` if they did)

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py [hunk 3]
```diff
@@ -7,18 +7,58 @@
...
+        :ivar image_format: The MIME type format of the image. For example: "jpeg", "png". Required.
+        :vartype image_format: str
+        :ivar detail: The evaluation quality setting to use, which controls relative prioritization of
+         speed, token consumption, and accuracy. Known values are: "auto", "low", and "high".
+        :vartype detail: str or ~azure.ai.inference.models.ImageDetailLevel
+        :return: An ImageUrl object with the image data encoded as a base64 string.
+        :rtype: ~azure.ai.inference.models.ImageUrl
+        :raises FileNotFoundError when the image file could not be opened.
+        """
+        with open(image_file, "rb") as f:
```
Comment (johanste) [inline sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py:56]: The main issue with opening a file inside a method is that it is a sync IO operation. And there is no async file `open` in the standard python library/builtins. We can either move the file open outside the method (and ultimately end up supporting both a sync and async file) or we can add a `async def aload(...)` method in the future.

I think the latter is fine (e.g. the code as written doesn't dig ourselves into too deep of a hole). But @annatisch , @kashifkhan , not sure if you have opinions on this?

**Review by johanste — APPROVED** [review]

File: sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py (hunk 3 above)
Comment (dargilco) [reply sdk/ai/azure-ai-inference/azure/ai/inference/models/_patch.py:56]: Per phone call, we can continue discussing this and make fixes as needed in another PR.

**Pull request merged**